teps/0025-proxy-support.md

```diff
@@ -0,0 +1,133 @@
+---
+title: proxy-support
+
+authors:
+  - "@piyush-garg"
+  - "@vdemeester"
+  
+creation-date: 2020-03-14
+
+last-updated: 2020-03-14
+
+status: proposed
+
+---
+
+# TEP-0025: Proxy Support
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (optional)](#user-stories-optional)
+  - [Notes/Constraints/Caveats (optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+The proposal is to provide the proxy support to tektoncd/pipelines and tektoncd/triggers.
+
+Tekton should provide a way to use network proxy settings defined for the platform so that workloads
+can make use of it during execution.
+
+## Motivation
+
+This will enable users/organisations to enable network proxy settings for CI/CD using tekton if their
+workload is expected to communicate to outside world through proxy.
+
+### Goals
+
+1. Workloads executions like taskrun/pipelinerun should use global proxy settings.
+2. Controllers should use global proxy settings for communicating.
+
+### Non-Goals
+
+1. Managing or maintaining proxy settings.
+
+## Proposal
+
+This proposal is to allow tektoncd/pipeline and tektoncd/triggers to have proxy support
+enabled / configurable to benefit users/organizations to use proxy feature to communicate to
+the outside world of the platform.
+
+It can be configured to use proxy setting defined in the platform.
+
+### User Stories (optional)
+
+User should be able to define global proxy settings on the platform and tektoncd/pipeline
+and tektoncd/triggers act based on the proxy settings defined.
+
+1. User should be able to define proxy config for all workloads
+2. User should be able to provide proxy config for specific taskrun/pipelinerun
+3. Controllers should use proxy setting for communication
```

Comment:
This should be under the `Requirements` header, this is not a user story.
A user story would be "User A has a http proxy setup on his cluster and …", or "User B has an http/https proxy setup on his cluster, with tls certs and …"